Add STAmount calc switchover support to tx queue

ximinez opened the pull request:
These commits will get squashed before merging, so no need to individually review. I just made some last minute tweaks and want to confirm CI still passes.

seelabs commented on src/test/app/Flow_test.cpp:
```diff
@@ -1282,10 +1282,19 @@ struct Flow_test : public beast::unit_test::suite
         testWithFeats();
+        // Don't bother testing the old payment engine with the queue,
+        // because it's old and buggy. That's why we updated it.
+        //testWithFeats(featureFeeEscalation);
```
I'd just remove it all together rather than leave the dead code commented out.

ximinez replied:
Fixed.

seelabs commented:
This is transaction breaking and needs an amendment.

ximinez commented:
TODO: change the std::unique_ptr in that last commit to boost::optional

ximinez commented:
The new functionality in `TxQ` is now behind an amendment. I also updated the `Offer` and `Flow` tests so they can run with (in principal) any combo of amendments, but updated to only run with the most realistic combo (fee escalation and the new fix1513). Finally, I added `Offer_manual` and `Flow_manual` manual tests with all the relevant combos of amendments because they take way too long to run every time.

ximinez commented:
And rebased again up to rc1.

seelabs commented on src/test/app/Flow_test.cpp:
```diff
@@ -1254,6 +1254,34 @@ struct Flow_test : public beast::unit_test::suite
     }
 
+    template<class... T>
+    void testWithFeats(bool haveFlow, T&&... fs)
```
We can get rid of the `bool` parameter and check for `haveBool` directly in the param pack, but it requires the changes in this PR: https://github.com/ripple/rippled/pull/2216

I'd like to make that change, but I think the easiest is to wait for both PRs to be merged and to created another PR later with the change.

seelabs commented on src/test/app/Flow_test.cpp:
```diff
@@ -1262,35 +1290,50 @@ struct Flow_test : public beast::unit_test::suite
+struct Flow_manual_test : public Flow_test
+{
+    void run() override
+    {
```
I like this change a lot. Good thinking!

ximinez replied:
Thanks!

seelabs commented on src/test/app/Flow_test.cpp:
```diff
@@ -1254,6 +1254,34 @@ struct Flow_test : public beast::unit_test::suite
+    template<class... T>
+    void testWithFeats(T&&... fs)
+    {
+        testWithFeats(!!sizeof...(fs), fs...);
```
I prefer `sizeof...(fs) > 0` or `sizeof...(fs) != 0` to `!!`. But that's clearly just a personal preference, no change needed. Plus this code will go away later anyway.

ximinez replied:
I'll leave it for later, then.

ximinez commented:
* Rebased on 0.80.0-rc2
* Resolved conflicts
* Squashed
* Updated the commit message (@seelabs @miguelportilla Could you check the commit message out?)

seelabs commented:
Commit message looks good to me. If I were to nit, I'd probably call it `STAmount switchover` and leave out the `calc`, but it really doesn't matter. Fine as is.

ximinez commented:
I'll change it since I realized I forgot the `:`

seelabs commented:
In 0.90.0-b1
